src/lib/HST/Frontend/FromHSE.hs

```diff
@@ -25,201 +44,437 @@ data HSE
 type instance S.SrcSpanType HSE = HSE.SrcSpanInfo
 type instance S.Literal HSE = HSE.Literal HSE.SrcSpanInfo
 type instance S.TypeExp HSE = HSE.Type HSE.SrcSpanInfo
+type instance S.OriginalModuleHead HSE = OriginalModuleHead
+type instance S.OriginalDecl HSE = HSE.Decl HSE.SrcSpanInfo
 
 instance S.EqAST HSE
 instance S.ShowAST HSE
 
+------------------------------------------------------------------------------
+-- Error messages                                                           --
+------------------------------------------------------------------------------
+
+-- | Reports a fatal error that the given feature is not supported.
+notSupported
+  :: Member Report r
+  => String -- ^ The name of the feature (plural) that is not supported.
+  -> Sem r b
+notSupported feature =
+  reportFatal $ Message Error $ feature ++ " are not supported!"
+
+-- | Informs the user that the given feature is not supported and the
+--   corresponding AST node will be skipped.
+skipNotSupported
+  :: Member Report r
+  => String -- ^ The name of the feature (plural) that is not supported.
+  -> Sem r ()
+skipNotSupported feature =
+  report $ Message Info $ feature ++ " are not supported and will be skipped!"
```

Comment:
These functions will be identical for the GHC transformation. I will copy them for now into `FromGHC.hs`, so I can start working with using `report` instead of `error` in the GHC modules, but they should be moved to another module, maybe `HST.Effect.Report`.

Reply:
See also #40.